to_copy/usr/bin/obtain-cert-end-date.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash -u
+
+
+
+# An OpenVPN tls-verify script
+#
+# This would obtain certificate's end date of a pending TLS connection.
+
+
+LOCKFILE="/etc/encryptme/data/.cert_lock"
+END_DATE_FILE="/etc/encryptme/data/cert_end_date"
+
+fail() {
+    echo "${1-command failed}" >&2
+    [ -f "$LOCKFILE" ] && rm $LOCKFILE 
+    exit ${2:-1}
+}
+
+depth="$1"
+subject="$2"
+
+count=0
+while [ -f "$LOCKFILE" ]; do
+    sleep 1
+    ((count++))
+    [ $count -gt 5 ] && fail "Could not get a lock"
+done
+echo "$$" > $LOCKFILE
+
+# Verify to only use the end-entity certificate in the chain, discarding CA
+# certificates
+[ "$depth" -eq 0 ] && {
+
+    end_date=$(openssl x509 -noout -in "$peer_cert" -enddate | cut -d "=" -f 2)
```

Comment:
$peer_cert it's used before defined?

Reply:
This is a environmental variable set by OpenVPN